src/NextGen/DonationForm/DataTransferObjects/DonateControllerData.php

```diff
@@ -179,10 +205,40 @@ public function getCustomFields(): array
                         'isEmbed',
                         'embedId',
                         'donationType',
+                        'subscriptionPeriod',
+                        'subscriptionFrequency',
+                        'subscriptionInstallments'
                     ]
                 ),
                 true
             );
         }, ARRAY_FILTER_USE_KEY);
     }
+
+    /**
+     * @unreleased
+     *
+     * @throws Exception
+     */
+    public function toSubscription(int $donorId): Subscription
+    {
+        return new Subscription([
+            'amount' => $this->amount(),
+            'period' => $this->subscriptionPeriod,
+            'frequency' => (int)$this->subscriptionFrequency,
+            'donorId' => $donorId,
+            'installments' => (int)$this->subscriptionInstallments,
+            'status' => SubscriptionStatus::PENDING(),
+            'mode' => give_is_test_mode() ? SubscriptionMode::TEST() : SubscriptionMode::LIVE(),
+            'donationFormId' => $this->formId,
+        ]);
+    }
+
+    /**
+     * @return Money
+     */
+    public function amount(): Money
```

Comment:
Looks like this can be private?

Reply:
I'm okay with this being public, it's just a helper for formatting two public properties. Could be useful in the controller actions